### ISSUE TITLE
snmpd ifName support

### ISSUE THREAD

**Issue by nvx** [netdata/netdata#1164]
Currently snmp support in netdata relies on hardcoding OIDs, which use the ifIndex for network interfaces.

This is a problem as ifIndexes are not required to be unique according to the RFC, and on a number of systems (notably anything Linux derived) they change.

Being able to reference an interface by ifName or similar would address this.

**Comment by ktsaou** [comment]
hm... nice request, thank you.

I have to check how this can be done with net-snmp on node.js...

**Comment by magao** [comment]
I've just encountered this while trying to monitor my VPN tunnels. I've had a look at the code of the SNMP plugin but it's currently a bit impenetrable to me.

What I'm thinking is being able to specify a table OID, a sub-OID for the value, and a sub-OID to lookup a value to determine the index. First the lookup tree would be retrieved and the varbinds searched for a matching value. The found sub-OID would then be appended to the value sub-OID to generate the actual OID with the required value.

So where I've currently got:

```
"vpn.bandwidth": {
    "title": "VPN Bandwidth",
    "units": "kilobits/s",
    "type": "area",
    "priority": 1,
    "family": "ports",
    "dimensions": {
        "received": {
            "oid": "1.3.6.1.2.1.2.2.1.10.6",
            "algorithm": "incremental",
            "multiplier": 8,
            "divisor": 1024,
            "offset": 0
        }
    }
}
```

I could instead specify:

```
"vpn.bandwidth": {
    "title": "VPN Bandwidth",
    "units": "kilobits/s",
    "type": "area",
    "priority": 1,
    "family": "ports",
    "dimensions": {
        "received": {
            "oid": {
                "table-oid": "1.3.6.1.2.1.2.2.1",
                "value-suboid": "10",
                "lookup-suboid": "2",
                "lookup-match": "tun0"
            }
            "algorithm": "incremental",
            "multiplier": 8,
            "divisor": 1024,
            "offset": 0
        }
    }
}
```

and it would retrieve the `1.3.6.1.2.1.2.2.1.2` sub-tree, determine that `1.3.6.1.2.1.2.2.1.2.6` matched the value `tun0` and then retrieve `1.3.6.1.2.1.2.2.1.10.6`.

**Comment by magao** [comment]
Thinking about this further, I think the optimal strategy would be:

1. If we haven't yet identified the ifIndex, retrieve the lookup subtree and find it.

2. Once we've identified the ifIndex, retrieve both the lookup OID and the value OID for that ifIndex (in one call if possible).

3. If the lookup varbind still matches the lookup value, return the retrieved value, otherwise forget the ifIndex (forcing a search through the subtree next time).

I'm much more comfortable with Python than Node.js (which I have zero experience with) so I might try rewriting the plugin in Python using [pysnmp](http://pysnmp.sourceforge.net/) as discussed in #2496. But I've actually got things working from the other end using statsd for my VPN tunnels so no urgency on my part ...
